sw/device/tests/sim_dv/spi_passthrough_test.c

```diff
@@ -0,0 +1,419 @@
+      },
+  };
+  for (int i = 0; i < ARRAYSIZE(write_commands); ++i) {
+    uint8_t slot = i + kSpiDeviceWriteCommandSlotBase;
+    CHECK_DIF_OK(dif_spi_device_set_flash_command_slot(
+        &spi_device, slot, kDifToggleEnabled, write_commands[i]));
+  }
+  CHECK_DIF_OK(dif_spi_device_configure_flash_wren_command(
+      &spi_device, kDifToggleEnabled, kSpiFlashOpWriteEnable));
+  CHECK_DIF_OK(dif_spi_device_configure_flash_wrdi_command(
+      &spi_device, kDifToggleEnabled, kSpiFlashOpWriteDisable));
+  CHECK_DIF_OK(dif_spi_device_configure_flash_en4b_command(
+      &spi_device, kDifToggleEnabled, kSpiFlashOpEnter4bAddr));
+  CHECK_DIF_OK(dif_spi_device_configure_flash_ex4b_command(
+      &spi_device, kDifToggleEnabled, kSpiFlashOpExit4bAddr));
+}
+
+bool test_main(void) {
+  // Initialize the pinmux.
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  pinmux_testutils_init(&pinmux);
+  for (int i = 0; i < ARRAYSIZE(pinmux_pad_config); ++i) {
+    dif_pinmux_pad_attr_t attr, attr_check;
+    pinmux_pad_attributes_t config = pinmux_pad_config[i];
+    CHECK_DIF_OK(
+        dif_pinmux_pad_get_attrs(&pinmux, config.pad, config.kind, &attr));
+    attr.flags = config.flags;
+    CHECK_DIF_OK(dif_pinmux_pad_write_attrs(&pinmux, config.pad, config.kind,
+                                            attr, &attr_check));
+    // Check that attributes were accepted?
+  }
+  for (int i = 0; i < ARRAYSIZE(pinmux_in_config); ++i) {
+    pinmux_select_t setting = pinmux_in_config[i];
+    CHECK_DIF_OK(
+        dif_pinmux_input_select(&pinmux, setting.peripheral, setting.pad));
+  }
+  for (int i = 0; i < ARRAYSIZE(pinmux_out_config); ++i) {
+    pinmux_select_t setting = pinmux_out_config[i];
+    CHECK_DIF_OK(
+        dif_pinmux_output_select(&pinmux, setting.pad, setting.peripheral));
+  }
+
+  // Initialize the spi_host devices.
+  CHECK_DIF_OK(dif_spi_host_init(
+      mmio_region_from_addr(TOP_EARLGREY_SPI_HOST0_BASE_ADDR), &spi_host0));
+  CHECK_DIF_OK(dif_spi_host_init(
+      mmio_region_from_addr(TOP_EARLGREY_SPI_HOST0_BASE_ADDR), &spi_host1));
+  init_spi_host(&spi_host0, (uint32_t)kClockFreqHiSpeedPeripheralHz);
+  init_spi_host(&spi_host1, (uint32_t)kClockFreqPeripheralHz);
+
+  // Initialize spi_device.
+  mmio_region_t spi_device_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_SPI_DEVICE_BASE_ADDR);
+  CHECK_DIF_OK(dif_spi_device_init_handle(spi_device_base_addr, &spi_device));
+  init_spi_device();
+
+  // Send the DV environment a specific message for synchronization. The
+  // sequencer can pick this up and know it can begin sending SPI flash
+  // transactions.
+  LOG_INFO("Test setup complete.");
+
+  // Stop here, as the DV environment is the only entity that can determine if
+  // the test has passed.
+  wait_for_interrupt();
+
+  // No interrupts are expected, so fail if one occurred.
+  LOG_INFO("Received unexpected interrupt, ending wfi");
```

Comment:
i might have missed it, but i dont think we actually enabled interrupts in this test right?

Reply:
That's right, so this is merely information about an unusual termination.

Reply:
do you want to add the irq enabling? Especially if we expect them to NOT fire in this test, it might be a good negative check.

Reply:
Which interrupts did you have in mind? I feel a bit ambivalent here, heh.

Reply:
ah for some of my tests (especially after the i2c unexpected error experience) i just started enabling all interrupts that i don't expect to trigger...